packages/elements/src/components/file/file.component.ts

```diff
@@ -1,82 +1,168 @@
 import { unsafeHTML } from 'lit/directives/unsafe-html.js';
-import { MutationTestReportMutantComponent } from '../mutant/mutant.component';
 import { StateFilter } from '../state-filter/state-filter.component';
 import { bootstrap, prismjs } from '../../style';
-import { markMutants } from '../../lib/code-helpers';
-import { FileResult, MutantStatus } from 'mutation-testing-report-schema/api';
-import { highlightElement } from 'prismjs/components/prism-core';
+import { findDiffIndices, gte, highlightCode, transformHighlightedLines } from '../../lib/code-helpers';
+import { MutantResult, MutantStatus } from 'mutation-testing-report-schema/api';
 import style from './file.scss';
-import { getContextClassForStatus, getEmojiForStatus } from '../../lib/htmlHelpers';
+import { getContextClassForStatus, getEmojiForStatus, scrollToCodeFragmentIfNeeded } from '../../lib/html-helpers';
+import { FileUnderTestModel, MutantModel } from 'mutation-testing-metrics';
+import { createCustomEvent, MteCustomEvent } from '../../lib/custom-events';
 
+const diffOldClass = 'diff-old';
+const diffNewClass = 'diff-new';
 @customElement('mte-file')
-export class MutationTestReportFileComponent extends LitElement {
-  @property()
-  private filters: StateFilter<MutantStatus>[] = [];
+export class FileComponent extends LitElement {
+  @state()
+  public filters: StateFilter<MutantStatus>[] = [];
 
   @property()
-  public model!: FileResult;
+  public model!: FileUnderTestModel;
 
-  public static styles = [prismjs, bootstrap, unsafeCSS(style)];
+  @state()
+  public selectedMutantStates: MutantStatus[] = [];
 
-  private codeRef = createRef();
+  @state()
+  private selectedMutant?: MutantModel;
 
-  private readonly expandAll = () => {
-    this.forEachMutantComponent((mutantComponent) => (mutantComponent.expand = true));
-  };
-  private readonly collapseAll = () => {
-    this.forEachMutantComponent((mutantComponent) => (mutantComponent.expand = false));
+  @state()
+  private lines: string[] = [];
+
+  @state()
+  public mutants: MutantModel[] = [];
+
+  public static styles = [prismjs, bootstrap, unsafeCSS(style)];
+  private codeRef = createRef<HTMLElement>();
+
+  private readonly filtersChanged = (event: MteCustomEvent<'filters-changed'>) => {
+    this.selectedMutantStates = event.detail as MutantStatus[];
   };
 
-  private forEachMutantComponent(action: (mutant: MutationTestReportMutantComponent) => void) {
-    for (const mutantComponent of this.shadowRoot!.querySelectorAll('mte-mutant')) {
-      if (mutantComponent instanceof MutationTestReportMutantComponent) {
-        action(mutantComponent);
+  private codeClicked = (ev: MouseEvent) => {
+    ev.stopPropagation();
+
+    if (ev.target instanceof Element) {
+      let maybeMutantTarget: Element | null = ev.target;
+      const mutantsInScope: MutantModel[] = [];
+      for (; maybeMutantTarget instanceof Element; maybeMutantTarget = maybeMutantTarget.parentElement) {
```

Comment:
Only the first iteration. That's why I'm declaring it as `Element | null` at line 47, otherwise `maybeMutantTarget = mayMutantTarget.parentElement` won't work. Does that clear it up?

`parentElement` is of type `Element | null` (if I'm not mistaken).